components/feature/recentlyclosed/src/test/java/mozilla/components/feature/recentlyclosed/RecentlyClosedMiddlewareTest.kt

```diff
@@ -82,8 +82,6 @@ class RecentlyClosedMiddlewareTest {
             store.dispatch(RecentlyClosedAction.AddClosedTabsAction(listOf(closedTab)))
                 .joinBlocking()
 
-            dispatcher.advanceUntilIdle()
```

Comment:
I think those calls made sense? In the middleware you are dispatching coroutines on the scope using that dispatcher. So you may want to wait for the to complete before you can verify that something happened after they executed. 🤔 

However some of the coroutines also dispatch something on the store. So you may also want to wait until the store has processed that (which also happens async). 

We added `Store.waitUntilIdle()` for doing that in tests.